lib/compress/zstd_compress.c

```diff
@@ -1317,18 +1379,24 @@ ZSTD_sizeof_matchState(const ZSTD_compressionParameters* const cParams,
       + ZSTD_cwksp_alloc_size((1<<Litbits) * sizeof(U32))
       + ZSTD_cwksp_alloc_size((ZSTD_OPT_NUM+1) * sizeof(ZSTD_match_t))
       + ZSTD_cwksp_alloc_size((ZSTD_OPT_NUM+1) * sizeof(ZSTD_optimal_t));
+    size_t const lazyAdditionalSpace = (useRowMatchfinder == ZSTD_urm_enableRowMatchfinder
```

Comment:
I don't really like the alignment code added. It feels like abstraction leak from `ZSTD_cwksp`. I think we should solve the alignment problem generically in a separate PR, and then stack on top of it (its fine to include that stack in this PR until it gets merged).

I think a good solution would be: Tables & aligned allocations are all aligned to 64 bytes. All the logic is handled in `ZSTD_cwksp`. We could get there by:
1. `ZSTD_cwksp_alloc_size()` gets broken into separate functions for each allocation type (they can all funnel to the same implementation internally). This will allow us to enforce that allocations in the table & and aligned sections are all multiples of 64-bytes.
2. Add a function `ZSTD_cwksp_slack_space()` that would return the number of slack bytes that need to be allocated for internal purposes (currently only alignment).
3. `ZSTD_cwksp_internal_advance_phase()` would handle the alignment at the beginning of the tables/aligned sections
4. `ZSTD_cwksp_reserve_table()` and `ZSTD_cwksp_reserve_aligned()` align their allocation sizes to a multiple of 64 (accounted for by the alloc size function).

This adds a little bit of complexity to `ZSTD_cwksp`, but now it is all wrapped in the abstraction, and `zstd_compress.c` doesn't need to know anything about it.